src/sentry/notifications/helpers.py

```diff
@@ -715,6 +719,51 @@ def get_recipient_from_team_or_user(user_id: int | None, team_id: int | None) ->
     return recipient
 
 
+def team_is_valid_recipient(team: Team | RpcActor) -> bool:
+    """
+    A team is a valid recipient if it has a linked integration (ie. linked Slack channel)
+    for any one of the providers allowed for personal notifications.
+    """
+    from sentry.models.integrations.external_actor import ExternalActor
+
+    linked_integration = ExternalActor.objects.filter(
+        team_id=team.id,
+        provider__in=PERSONAL_NOTIFICATION_PROVIDERS_AS_INT,
+    )
+    if linked_integration:
+        return True
+    return False
+
+
+def get_team_members(team: Team | RpcActor) -> list[RpcActor]:
+    if recipient_is_team(team):  # handles type error below
+        team_id = team.id
+    else:  # team is either Team or RpcActor, so if recipient_is_team returns false it is because RpcActor has a different type
+        raise Exception(
+            "RpcActor team has ActorType %s, expected ActorType Team", team.actor_type  # type: ignore
+        )
+
+    # get organization member IDs of all members in the team
+    team_members = OrganizationMemberTeamReplica.objects.filter(team_id=team_id)
+
+    # use the first member to get the org id + determine if there are any members to begin with
+    first_member = team_members.first()
+    if not first_member:
+        return []
+    org_id = first_member.organization_id
+
+    # get user IDs for all members in the team
+    members = OrganizationMemberMapping.objects.filter(
+        organization_id=org_id,
+        organizationmember_id__in=Subquery(team_members.values("organizationmember_id")),
+    )
+
+    return [
+        RpcActor(id=user_id, actor_type=ActorType.USER)
+        for user_id in members.values_list("user_id", flat=True)
```

Comment:
@isabellaenriquez we probably should check if `user_id` is null before we call `RpcActor` on it